certbot/main.py

```diff
@@ -600,25 +593,29 @@ def run(config, plugins):  # pylint: disable=too-many-branches,too-many-locals
     except errors.PluginSelectionError as e:
         return e.message
 
-    domains, certname = _find_domains_or_certname(config, installer)
-
     # TODO: Handle errors from _init_le_client?
     le_client = _init_le_client(config, authenticator, installer)
 
-    action, lineage = _auth_from_available(le_client, config, domains, certname)
+    domains, certname = _find_domains_or_certname(config, installer)
+    should_get_cert, lineage = _find_cert(config, domains, certname)
+
+    new_lineage = lineage
+    if should_get_cert:
+        new_lineage = _get_and_save_cert(le_client, config, domains,
+            certname, lineage)
 
-    le_client.deploy_certificate(
-        domains, lineage.privkey, lineage.cert,
-        lineage.chain, lineage.fullchain)
+    cert_path = new_lineage.cert_path if new_lineage else None
+    fullchain_path = new_lineage.fullchain_path if new_lineage else None
+    _report_new_cert(config, cert_path, fullchain_path)
 
-    le_client.enhance_config(domains, lineage.chain)
+    _install_cert(config, le_client, domains, new_lineage)
 
-    if action in ("newcert", "reinstall",):
+    if lineage is None or not should_get_cert:
         display_ops.success_installation(domains)
     else:
         display_ops.success_renewal(domains)
 
-    _suggest_donation_if_appropriate(config, action)
+    _suggest_donation_if_appropriate(config)
 
 
 def _csr_obtain_cert(config, le_client):
```

Comment:
nit: If we're renaming `obtain_cert` to `certonly`, should this be renamed to `_csr_certonly`?